src/ResolvedName.re

```diff
@@ -95,24 +95,38 @@ let rec extendExportModules = (x, ~exportModuleItems, ~valueName) =>
     }
   };
 
-let globalExportModuleItems: exportModuleItems = Hashtbl.create(1);
+type moduleItemsEmitter = Hashtbl.t(string, exportModuleItem)
+and moduleItem = (string, string);
+
+let createModuleItemsEmitter = () => Hashtbl.create(1);
+
+let moduleItemToString = ((moduleName, exportModuleItem)) =>
+  "export const " ++ moduleName ++ " = " ++ exportModuleItem ++ ";";
+
+let emitAllModuleItems = (~emitters, moduleItemsEmitter) => {
+  let emitters = ref(emitters);
 
-let emitAllModuleItems = () => {
-  let buffer = Buffer.create(0);
   Hashtbl.iter(
     (moduleName, exportModuleItem) => {
-      Buffer.add_string(buffer, "export const " ++ moduleName ++ " = ");
+      let buffer = Buffer.create(0);
       exportModuleItem |> emitExportModuleItem(~buffer);
-      Buffer.add_string(buffer, ";\n");
+
+      emitters :=
```

Comment:
is there a nicer way to iter trough this and same it into emitters ?

Reply:
Yes, one of the issues with hashtable iterators is that the order is unspecified.
So this is the right point to iterate once and build a list, then sort the list, and proceed with list fold.

Notice the alternative is to use maps instead of hash tables. The result is cleaner but the code more complex (need to use functors).